src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -97,13 +97,25 @@ private static bool IsCatchAll(string path, int counterForPath, string pathTempl
                      && pathTemplate.IndexOf('}') == pathTemplate.Length - 1;
         }
 
+        private static bool IsCatchAllAfterOtherPlaceholders(string pathTemplate, int counterForTemplate)
+            => (pathTemplate[counterForTemplate] == '/')
+                && (counterForTemplate < pathTemplate.Length - 1)
+                && (pathTemplate[counterForTemplate + 1] == '{')
+                && NoMoreForwardSlash(pathTemplate, counterForTemplate + 1)
+                && NotPassedQueryString(pathTemplate, pathTemplate.Length);
```

Comment:
2 tests for
- CatchAll at the end
- CatchAll at the end after a placeholder in the middle

Reply:
@raman-m added extra test cases in here https://github.com/ThreeMammals/Ocelot/pull/1911/files#diff-fe239522803966c25dc7eda0de976c6b65aedb1d095c709cf3e414aa06f62f4cR389

Reply:
> I'm sorry! What URL paths do we process here? Like this? 👇
> 
> ```
> "DownstreamPathTemplate": "/api/{version}/test/{url}"
> "UpstreamPathTemplate": "/test/{version}/{url}"
> ```
> 
> According to our [Placeholders](https://ocelot.readthedocs.io/en/latest/features/routing.html#placeholders:~:text=This%20will%20forward%20any%20path%20%2B%20query%20string%20combinations%20to%20the%20downstream%20service%20after%20the%20path%20%2Fapi.) docs and [Catch All](https://ocelot.readthedocs.io/en/latest/features/routing.html#id1) docs, we have to forward query string too ❗ **We must forward query string in case of empty Catch All placeholders too!** Seems we have to remove the line 105.
> 
> Could you add extra 1 unit and 1 acceptance tests which check the case with query string too, please?

we don't have to remove that line because this branch comes into effect when there is a specific condition of no query strings till the current PathCounter other than that it is handled normally but the if branch not the else if branch